contracts/MetadataViews.cdc

```diff
@@ -238,4 +239,63 @@ pub contract MetadataViews {
             self.url=url
         }
     }
+
+
+    // A view to expose the information needed store and retrieve an NFT
+    //
+    // This can be used by applications to setup a NFT collection with proper storage and public capabilities.
+    pub struct NFTCollectionData {
+        // Path in storage where this NFT is recommended to be stored.
+        pub let storagePath: StoragePath
+
+        // Public path which must be linked to expose public capabilities of this NFT
+        // including standard NFT interfaces and metadataviews interfaces
+        pub let publicPath: PublicPath
+
+        // Private path which should be linked to expose the provider
+        // capability to withdraw NFTs from the collection holding NFTs
+        pub let providerPath: PrivatePath
+
+        // Public collection type that is expected to provide sufficient read-only access to standard
+        // functions (deposit + getIDs + borrowNFT)
+        // This field is for backwards compatibility with collections that have not used the standard
+        // NonFungibleToken.CollectionPublic interface when setting up collections. For new
+        // collections, this may be set to be equal to the type specified in `publicLinkedType`.
+        pub let publicCollection: Type
+
+        // Type that should be linked at the aforementioned public path. This is normally a
+        // restricted type with many interfaces. Notably the `NFT.CollectionPublic`,
+        // `NFT.Receiver`, and `MetadataViews.ResolverCollection` interfaces are required.
+        pub let publicLinkedType: Type
+
+        // Type that should be linked at the aforementioned private path. This is normally
+        // a restricted type with at a minimum the `NFT.Provider` interface
+        pub let providerLinkedType: Type
+
+        // Function that allows creation of an empty NFT collection that is intended to store
+        // this NFT.
+        pub let createEmptyCollectionFunction: ((): @NonFungibleToken.Collection)
```

Comment:
Small nitpick, but I'd probably call this `createEmptyCollection` just so that it reads more naturally when invoked:

```cadence
let collection = data.createEmptyCollection()
```

The init parameter could still be `createEmptyCollectionFunction` though

Reply:
I like that, updated to match the suggestion.